### PR TITLE
 #32 Added Apache access.log like logger for 2 endpoints

### LINKED ISSUE #32: Generate query audit log
Update the backend so that whenever a query is run, information is saved to the file `access.log`. This file will be similar to an Apache access log file and should include the following information:
- current timestamp
- logged in user
- query URL
- query post body params
- query success/failure(status code)
- time taken for query response

A log line will need to be added whenever the `/api/search` or `/api/search/ws` endpoints are accessed. The `pkg/server/query/server.go` file shows which functions are called when these endpoints are accessed.

### PR DESCRIPTION
**Pull request by girish332**
# Description

Added apache access.log like functionality to 2 endpoints : 
1) /api/search
2) /api/search/ws

Fixed #32 

# Testing
Added unit tests for access.log file 
Sample Log present in access.log file is : 

```2023-11-21 17:21:16 No-user http:/// {"indexName":"traces","searchText":"","startEpoch":"now-5m","endEpoch":"now","queryLanguage":"Splunk QL"} 200 0```

# Checklist:
Before marking your pull request as ready for review, complete the following.

- [x] I have self-reviewed this PR.
- [x] I have removed all print-debugging and commented-out code that should not be merged.
- [x] I have added sufficient comments in my code, particularly in hard-to-understand areas.
- [x] I have formatted the code, if applicable. For Go, I have run `goimports -w .`.

### CODE REVIEW

**Review by AndrewHess — COMMENTED** [review]
So far this looks really good

**Review by AndrewHess — COMMENTED** [review]
Looks good. The `UserName` will be implemented in a different PR after some other changes are made to handle user auth, right? So this PR is ready to get merged?

**Comment by nkunal** [comment]
> Looks good. The `UserName` will be implemented in a different PR after some other changes are made to handle user auth, right? So this PR is ready to get merged?

Yes that's correct

**Comment by sunitakawane** [comment]
Some of entries in access.log are related to internal queries we run periodically for traces related data. These searchRequests have no URL(http:///), and get called from MakeTracesDependancyGraph and ProcessRedTracesIngest. We should not log these requests. Lets add a check if `indexName"="traces"`, do not log the search requst. @nkunal wdyt

**Comment by nkunal** [comment]
> Some of entries in access.log are related to internal queries we run periodically for traces related data. These searchRequests have no URL(http:///), and get called from MakeTracesDependancyGraph and ProcessRedTracesIngest. We should not log these requests. Lets add a check if `indexName"="traces"`, do not log the search request. @nkunal wdyt

yes we should not log our internal requests. However we should log the requests that come from the end user via our UI. So just filtering by `indexName=traces` is not a good idea, since the UI might send requests that will contain this indexname. We need to find another way to do this.

**Comment by nkunal** [comment]
however if rest of the things in this PR look good, then we can merge this one and @girish332 should submit another issue and create a seperate PR for the internal  requests

**Comment by sunitakawane** [comment]
Lets add check for `indexName"="traces"` and  `URL= http:///`, which will filter internal searchRequests for traces. We should not add these to access.log

**Comment by girish332** [comment]
I have made the required changes, kindly review them

File: pkg/utils/accessLogUtils.go [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+	if err != nil {
+		log.Errorf("Unable to write to access.log file, err=%v", err)
+	}
+	defer logFile.Close()
+	// Do not log websocket connections
+	if data.StatusCode == 101 {
+		return
+	}
+	// Do not log internal search requests for trace data
+	if strings.TrimSpace(data.URI) == "http:///" || strings.TrimSpace(data.URI) == "https:///" {
```
Comment (sunitakawane) [inline pkg/utils/accessLogUtils.go:23]: combine these two if loops  , we do not want to log queries with "http:///" AND indexName=traces

Comment (girish332) [reply pkg/utils/accessLogUtils.go]: Have made the changes

**Pull request merged**